Xamarin.Forms.Core/Shell/ShellUriHandler.cs

```diff
@@ -245,21 +249,6 @@ internal static List<RouteRequestBuilder> GenerateRoutePaths(Shell shell, Uri re
 
 			var segments = RetrievePaths(localPath);
 
-			if (!relativeMatch)
```

Comment:
This code just moved to the bottom of this method so we can try to fulfill the route before throwing an exception